src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2448,6 +2448,15 @@ var projectUniqueNamesForBuildIntToUpdate
                     token);
             }
 
+            // If HideWarningsAndErrors is true then restore will not display the warnings and errors.
+            // Further, replay errors and warnings only if restore failed because the assets file will not be committed.
+            // If there were only warnings then those are written to assets file and committed. The design time build will replay them.
+            if (updatedPackageSpec.RestoreSettings.HideWarningsAndErrors &&
+                !restoreResult.Result.Success)
```

Comment:
I saw @emgarten suggested this, but I don't get how this could lead to duplicates. Can you please explain?